File: sherpout-server/src/main/java/com/sherpout/server/api/exercise/enumerated/MuscleCategory.java
package com.sherpout.server.api.exercise.enumerated;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public enum MuscleCategory {
    CHEST("chest", "klata"),
    BACK("back", "plecy"),
    DELTS("delts", "barki"),
    ARMS("arms", "ramiona"),
    LEGS("legs", "nogi"),
    ABS("abs", "brzuch");

    private final String en;
    private final String pl;
}

File: sherpout_mobile/lib/common/dto/muscle_category.dart
import 'package:flutter/cupertino.dart';

enum MuscleCategory {
  chest('chest', 'klata'),
  back('back', 'plecy'),
  delts('delts', 'barki'),
  arms('arms', 'ramiona'),
  legs('legs', 'nogi'),
  abs('abs', 'brzuch');

  final String en;
  final String pl;

  const MuscleCategory(this.en, this.pl);

  static MuscleCategory? fromName(String name) {
    return MuscleCategory.values.firstWhere(
      (e) => e.name.toLowerCase() == name.toLowerCase(),
    );
  }

  String localized(BuildContext context) {
    final lang = Localizations.localeOf(context).languageCode;
    switch (lang) {
      case 'pl':
        return pl;
      case 'en':
      default:
        return en;
    }
  }
}
